src/utils/hooks/useGetResponsesAndElementState.tsx

```diff
@@ -71,7 +71,7 @@ const useGetResponsesAndElementState = (props: useGetResponsesAndElementStatePro
       const code = response?.templateElement?.code
       if (code) {
         currentResponses[code] = response?.value?.text
-        currentFullResponses[code] = response?.value
+        currentFullResponses[code] = { isValid: response?.isValid, ...response?.value }
```

Comment:
Is this to make sure `isValid` isn't undefined? I wonder if there is a way to define that the return is never undefined.

Reply:
No, it's so that the "Full" responses object has the full Response shape (in an object, `text` field, etc.) wheras the "response" just has the text string.

I think we're planning to refactor this at some point to only have 2 objects returned from this method rather than 3 at some point (#130) so this can be tidied up/got rid of in there.

Reply:
Ok. No problem. I was wondering what not just `currentFullResponses[code] = response` instead. But don't worry about this for now since it will be refactored soon : )